improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -972,26 +992,24 @@ def process(self, wet_bulb_temperature, wet_bulb_integral, orog,
         land_sea_data = next(land_sea_mask.slices([y_coord, x_coord])).data
         max_nbhood_orog = self.find_max_in_nbhood_orography(orography)
 
-        output_cube_name = 'altitude_of_{}_level'.format(
-            self.phase_change_name)
-        output_attributes = generate_mandatory_attributes(
-            [wet_bulb_temperature])
-
-        phase_change = iris.cube.CubeList([])
+        phase_change = None
         slice_list = ['height', y_coord, x_coord]
         for wb_integral, wet_bulb_temp in zip(
                 wet_bulb_integral.slices(slice_list),
                 wet_bulb_temperature.slices(slice_list)):
-
             phase_change_data = self._calculate_phase_change_level(
                 wet_bulb_temp.data, wb_integral.data, orography.data,
                 max_nbhood_orog.data, land_sea_data, wbt_height_points,
                 wb_integral.coord('height').points, highest_height)
 
-            phase_change_cube = create_new_diagnostic_cube(
-                output_cube_name, 'm', wb_integral[0], output_attributes,
-                data=phase_change_data)
-            phase_change.append(phase_change_cube)
+            # preserve dimensionality of input cube (in case of scalar or
+            # length 1 dimensions)
+            if phase_change:
```

Comment:
This line is failing the tests. Try something like
```suggestion
            if phase_change is None:
```

Reply:
I noticed.  It got messy.  Latest commit fixes this issue.